flang-rt/cmake/modules/HandleLibs.cmake

```
@@ -0,0 +1,50 @@
+#===-- cmake/modules/HandleLibs.cmake --------------------------------------===#
+#
+# Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+#===------------------------------------------------------------------------===#
+
+# Select the C library to use for building flang-rt.
+if (FLANG_RT_LIBC_PROVIDER STREQUAL "system")
+  add_library(flang-rt-libc-headers INTERFACE)
+  add_library(flang-rt-libc-static INTERFACE)
+  add_library(flang-rt-libc-shared INTERFACE)
+elseif (FLANG_RT_LIBC_PROVIDER STREQUAL "llvm")
+  add_library(flang-rt-libc-headers INTERFACE)
+  target_link_libraries(flang-rt-libc-headers INTERFACE libc-headers)
+  if (FLANG_RT_HAS_NOSTDLIBINC_FLAG)
+    target_compile_options(flang-rt-libc-headers INTERFACE $<$<COMPILE_LANGUAGE:CXX>:-nostdlibinc>)
```

```suggestion
    target_compile_options(flang-rt-libc-headers INTERFACE $<$<COMPILE_LANGUAGE:CXX,C>:-nostdlibinc>)
```
There is at least on `.c` source file, and C is used as LINKER_LANGUAGE